Tests/ReplaySubjectTests.swift

```diff
@@ -341,4 +317,30 @@ final class ReplaySubjectTests: XCTestCase {
 
         XCTAssertTrue(completed)
     }
+
+    func testDoubleSubscribe() {
+        let subject = ReplaySubject<Int, Never>(bufferSize: 1)
+
+        subject.send(1)
+        subject.send(2)
+        subject.send(completion: .finished)
+
+        var results = [Int]()
+        var completions = [Subscribers.Completion<Never>]()
+
+        let subscriber = AnySubscriber<Int, Never>(
+            receiveSubscription: { $0.request(.max(1)) },
+            receiveValue: { results.append($0); return .none },
+            receiveCompletion: { completions.append($0) }
+        )
+
+        subject
+            .subscribe(subscriber)
+
+        subject
+            .subscribe(subscriber)
+
+        XCTAssertEqual([2, 2], results)
+        XCTAssertEqual([.finished, .finished], completions)
```

Comment:
Perfect! The only change I'd make is give a salt here so you can actually identify these are two different subjects and not two emissions of the same subject, i.e.:


```suggestion
        subject
            .map { "a\($0)")
            .subscribe(subscriber)

        subject
            .map { "b\($0)") 
            .subscribe(subscriber)

        XCTAssertEqual(["a2", "b2"], results)
        XCTAssertEqual([.finished, .finished], completions)
```